src/AspNet.Security.OAuth.Introspection/OAuthIntrospectionHandler.cs

```diff
@@ -22,57 +22,114 @@
 
 namespace AspNet.Security.OAuth.Introspection {
     public class OAuthIntrospectionHandler : AuthenticationHandler<OAuthIntrospectionOptions> {
-        protected override async Task<AuthenticateResult> HandleAuthenticateAsync() {
-            string header = Request.Headers[HeaderNames.Authorization];
-            if (string.IsNullOrEmpty(header)) {
-                return AuthenticateResult.Fail("Authentication failed because the bearer token " +
-                                               "was missing from the 'Authorization' header.");
-            }
+        protected override async Task<AuthenticateResult> HandleAuthenticateAsync()
+        {
+            // Give the application an opportunity to parse the token from a different location, adjust, or reject token
+            var accessTokenReceivedContext = new AccessTokenReceivedContext(Context, Options);
```

Comment:
If so, we can't return the multiple failure messages while doing the parsing, and if they're not using the header at all it makes sense (at least to me) to skip that processing entirely.